# **2023**

## ***Jun 8 2023***

Энергия Kch для одного захода без точки пучка в фите и с ней:
<div>
<img src="PictureDump/KchEnergy_NoBeamPoint_run61683.png" alt="Drawing" style="width: 550px;"/>
<img src="PictureDump/KchEnergy_WithBeamPoint_run61683.png" alt="Drawing" style="width: 550px;"/>
</div>

Импульс Kch для одного захода без точки пучка в фите и с ней:
<div>
<img src="PictureDump/KchMom_NoBeamPoint_run61683.png" alt="Drawing" style="width: 550px;"/>
<img src="PictureDump/KchMom_WithBeamPoint_run61683.png" alt="Drawing" style="width: 550px;"/>
</div>

## ***Jun 9 2023***

MC KsKl with shifted up by 30 keV energy (aka WrongEnergy) Mass:
<div>
<img src="PictureDump/KsMass_WrongEnergy.png" alt="Drawing" style="width: 700px;"/>
</div>

In [11]:
import matplotlib.pyplot as plt
import numpy as np
from scipy.optimize import curve_fit

E = np.array([508.5, 509, 509.5, 510, 510.5, 511])
M_RCNC = np.array([497.575, 497.582, 497.588, 497.588, 497.588, 497.593])
M_RCNC_err = np.array([0.002, 0.002, 0.003, 0.003, 0.003, 0.003])

M_fit = curve_fit(lambda x, a: a, E, M_RCNC, sigma=M_RCNC_err)
print("M_fit =", np.round(M_fit[0][0], 3), "MeV")

# plt.errorbar(x=E, y=M_RCNC, yerr=M_RCNC_err, fmt='.')
# plt.show()

M_fit = 497.584 MeV


## ***Jun 10 2023***

Energy control via $K^{\pm}$ without beam point:

In [ ]:
energy_points = np.array([510])
E_Kch_exp = np.array([])
E_Kch_exp_Err = np.array([])

emeas_MC = np.array([509.95])
emeas_MC_err = np.array([0.007])
E_Kch_MC = np.array([506.249])
E_Kch_MC_Err = np.array([0.005])

## ***Jun 13 2023***

$\Delta E$ data - MC with and without beam point:
<div>
<img src="PictureDump/deltaE_dataMC_diff_withBeamPoint.png" alt="Drawing" style="width: 550px;"/>
<img src="PictureDump/deltaE_dataMC_diff_noBeamPoint.png" alt="Drawing" style="width: 550px;"/>
</div>

## ***Jun 14***

ISR radiative correction pretty much has no difference with or without energy spread. Previous RC is still good.

In [2]:
import numpy as np

energies = np.array([508, 508.5, 510])
RC_MC = np.array([0.082, 0.078, 0.106])
RC_smeared_MC = np.array([0.084, 0.078, 0.110])

m = np.array(
    [497.577, 497.568, 497.533, 497.554, 497.544, 497.57, 497.589, 497.568, 497.604]
)
RC = np.array([0.1, 0.099, 0.089, 0.08, 0.071, 0.116, 0.191, 0.334, 1.453])
print(list(np.round(m + RC, 3)))

[497.677, 497.667, 497.622, 497.634, 497.615, 497.686, 497.78, 497.902, 499.057]


I've started working on PhiMesonFit. 
There are some params for $e^+ e^- \rightarrow \phi \rightarrow K_S K_L$ for RHO2018 season.

## ***Jun 19***

Trying to *guess* the $\Delta E$ by minimizing spread (relative to mean value) of $M_{vis}(E) - \Delta M_{RC}(E + \Delta E)$:

We think $\Delta E = 96$ keV

In [6]:
import numpy as np
from matplotlib import pyplot as ppt

import sys
sys.path.append('C:\\work\\Science\\BINP\\Kaon Mass Measure')
from AuxScripts.EnergyShift import Shift

    
M_RCNC = np.array([497.577, 497.568, 497.533, 497.554, 497.544, 497.57, 497.589, 497.568, 497.604])
err_M = np.array([0.036, 0.014, 0.011, 0.008, 0.007, 0.01, 0.013, 0.014, 0.029])
# err_M = np.array([0.028, 0.01, 0.004, 0.004, 0.002, 0.002, 0.004, 0.007, 0.027])
RC_data = np.array([0.1, 0.099, 0.089, 0.08, 0.071, 0.116, 0.191, 0.334, 1.453])
E = np.array([504.8, 507.862, 508.404, 508.957, 509.528, 509.956, 510.458, 511.035, 513.864])
M_vis = np.round(M_RCNC + RC_data, 3)

shift = Shift(M_vis, err_M, RC_data, E)

deltaE, deltaE_err = shift.Get_Shift()
print("$\Delta E=$", deltaE, "+/-", deltaE_err, "MeV")

M_fit, err = shift.Get_Mass_fit()
print("M_fit =", M_fit, "+/-", err, "MeV")
print("M_new =", list(shift.Get_corrected_mass()))

$\Delta E=$ 0.096 +/- 0.033 MeV
M_fit = 497.55 +/- 0.004 MeV
M_new = [497.575, 497.57, 497.534, 497.557, 497.538, 497.557, 497.569, 497.541, 497.555]


## ***Jun 20***

In [46]:
import numpy as np

M_vis = np.array([497.766, 497.739, 497.706, 497.707, 497.711, 497.776, 497.855, 497.981, 499.148])
M_RCNC = np.round(M_vis - Get_RC(E + delta_E), 3)
print("M_RCNC =", list(M_RCNC))

M_fit, err = opt.curve_fit(lambda x, a: a, E, M_RCNC, sigma=err_M)
print("M_RCNC_fit =", np.round(M_fit[0], 3), "+/-", np.round(np.sqrt(np.diag(err))[0], 3), "MeV")

M_RCNC = [497.664, 497.642, 497.618, 497.63, 497.634, 497.647, 497.644, 497.619, 497.645]
M_RCNC_fit = 497.634 +/- 0.003 MeV


I splined ISR correction (Get_RC function) and fitted the function* to a M_vis data. 
So $\Delta E = 0.096 \pm 0.033$ MeV was obtained. 

*: M(E| $M_{mean}$, $\Delta E$) = $M_{mean}$ + Get_RC(E + $\Delta E$)

$M_{vis}$ (black), $\Delta M_{RC} + 497.550$ (blue) vs $E^{(mean)}_{compton}$
<div>
<img src="PictureDump/Mvis_RC.png" alt="Drawing" style="width: 850px;"/>
</div>

After 96 kev energy shift and applying corresponding ISR corrections: 

<div>
<img src="PictureDump/KsMassExp_CorrectedEnergy.png" alt="Drawing" style="width: 850px;"/>
</div>